readthedocs/projects/views/private.py

```diff
@@ -231,13 +231,22 @@ def done(self, form_list, **kwargs):
         for field, value in list(form_data.items()):
             if field in extra_fields:
                 setattr(project, field, value)
-        basic_only = True
         project.save()
+
+        # TODO: if we want to make the ``attach_webhook`` async, we need to
+        # consider the message shown to the user when not valid webhook.
         project_import.send(sender=project, request=self.request)
```

Comment:
I suppose that we can remove the receiver of this signal and use `django-async-messages` here also.

Reply:
We can achieve this in another PR, if you agree, after we test a little the new flow.

Reply:
Two things here:

1. I didn't know that we were using message extends
1. it doesn't do exactly the same since async message is a one-time message where the user doesn't have to interact with it (hit the close button) which sometimes could be lost by the user and don't read (it works exactly in the same way as normal django message framework does.

Summarizing, I think that what we were using (message extends) is better. I will adapt my code to use the Sticky storage.

Reply:
Mmm... I just read the code and found that it does _depend_ on the `request`.

I'm taking a look to see if we can extend the storage to just receive the user instead.

Reply:
I was thinking on something like this:

```python
from collections import namedtuple

from messages_extends.storages import PersistentStorage


class CeleryPersistentStorage(PersistentStorage):

    """
    Persistent Message Storage for Celery.

    This storage doesn't depend on a request as first argument.
    """

    FakeRequest = namedtuple('FakeRequest', ['user'])

    def __init__(self, user, *args, **kwargs):
        self.user = user
        request = self.FakeRequest(user=self.user)
        return super(CeleryPersistentStorage, self).__init__(request, *args, **kwargs)

    def get_user(self):
        return self.user
```

But I'm not sure if it's what we need and also it's a hack.

Reply:
I found another issue with `message_extends`.

In case we want to use it without a request, we can do this:

```python
from messages_extends import WARNING_PERSISTENT
from messages_extends.models import Message

Message.objects.create(
    user=self.user,
    message='Persistent message here',
    level=WARNING_PERSISTENT,
    expires=datetime,  # optional
)
```

This message will be displayed until it expires (if it's set). After it's expires it will be deleted from the database with a task that we have wrote.

The problem here is that we don't show the `X` to the user to close the message and be marked as read in the db (there are some `messages/` urls that we are not defining).

Reply:
I also found that we have an already coded hack here: https://github.com/rtfd/readthedocs.org/blob/31799754e46e7b51d8176686da0c34c4bded3917/readthedocs/notifications/backends.py#L81-L84

I will give it a try tomorrow. It seems that all I need is to call `notifications.backends.send_notification` with `None` as `request` and a level lower than `REQUIREMENT` to avoid sending the email and just show it in the site... (sounds tricky)